packages/tpetra/core/example/Finite-Element-Assembly/fem_assembly_InsertGlobalIndices_DP.cpp

```diff
@@ -58,19 +58,18 @@
 
 using namespace TpetraExamples;
 
+using comm_ptr_t = Teuchos::RCP<const Teuchos::Comm<int> >;
 
-int main (int argc, char *argv[]) 
+
+
+int execute(comm_ptr_t& comm)
```

Comment:
Next time, only pass by value or const reference (`const comm_ptr_t&`).  That conveys the intent not to reset the pointer.